### PR TITLE
Add clang-tidy PR comment bot

### PR DESCRIPTION
**Pull request by nsmithtt**

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -113,6 +113,22 @@ jobs:
...
+        # The GitHub token (or a personal access token)
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        # The path to the clang-tidy fixes generated above
+        clang_tidy_fixes: ${{ steps.strings.outputs.build-output-dir }}/clang-tidy-fixes.yml
+        # Optionally set to true if you want the Action to request
+        # changes in case warnings are found
+        request_changes: false
+        # Optionally set the number of comments per review
+        # to avoid GitHub API timeouts for heavily loaded
+        # pull requests
```
Comment (tapspatel) [inline .github/workflows/build-and-test.yml:128]: thanks for these comments. These workflow files have magic functions so I really like you commenting what it's doing :)

**Review by tapspatel — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (nsmithtt) [reply .github/workflows/build-and-test.yml:128]: I copy pasted them so I can't take credit :sweat_smile:

File: include/ttmlir/Dialect/TTNN/Transforms/Optimizer.h [hunk 2]
```diff
@@ -2,8 +2,8 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef TTMLIR_DIALECT_TTNN_TRANSFORMS_OPTIMIZER_H
-#define TTMLIR_DIALECT_TTNN_TRANSFORMS_OPTIMIZER_H
+#ifndef TTMLIR_DIALECT_TTNN_TRANSFORMS_OPTIMIZE_H
```
Comment (nobradovictt) [inline include/ttmlir/Dialect/TTNN/Transforms/Optimizer.h:5]: Why? :)

**Review by nobradovictt — APPROVED** [review]

File: include/ttmlir/Dialect/TTNN/Transforms/Optimizer.h (hunk 2 above)
Comment (nsmithtt) [reply include/ttmlir/Dialect/TTNN/Transforms/Optimizer.h:5]: Wait for the bot! :)

File: include/ttmlir/Dialect/TTNN/Transforms/Optimizer.h [hunk 3]
```diff
@@ -2,8 +2,8 @@
...
+#define TTMLIR_DIALECT_TTNN_TRANSFORMS_OPTIMIZE_H
```
[the remark this answers is not in the document]
Comment (nsmithtt) [reply include/ttmlir/Dialect/TTNN/Transforms/Optimizer.h:6]: @nobradovictt 

Now just to figure out why it's commenting multiple times for the same thing 😅

[the remark this answers is not in the document]
Comment (nsmithtt) [reply include/ttmlir/Dialect/TTNN/Transforms/Optimizer.h:6]: - you made a mistake
- you made a mistake
- you made a mistake
- you made a mistake
- you made a mistake
- you made a mistake
- you made a mistake

**Review by sdjordjevicTT — APPROVED** [review]

**Pull request merged**